examples/chemicalEnvironments/README.md

```diff
@@ -0,0 +1,177 @@
+For a more hands on tutorial you can look at the jupyter notebook `using_environments.py`. 
+
+### Initiating ChemicalEnvironments
+
+All Chemical Environments can be initated using SMIRKS strings. 
+The general ChemicalEnvironment, with no input string is completely empty, however there
+are 5 subtypes of ChemicalEnvironments that match the types of parameters found in the SMIRNOFF format:
+
+* AtomChemicalEnvironment
+    - expects 1 indexed atom
+    - default/generic SMIRKS `"[*:1]"`
+* BondChemicalEnvironment
+    - expects 2 indexed atoms
+    - default/generic SMIRKS: `"[*:1]~[*:2]"`
+* Angle
+    - expects 3 indexed atoms
+    - default/generic SMIRKS: `"[*:1]~[*:2]~[*:3]"`
+* Torsion
+    - expects 4 indexed atoms in a proper dihedral angle
+    - default/generic SMIRKS: `"[*:1]~[*:2]~[*:3]~[*:4]"`
+* Improper 
+    - expects 4 indexed atoms in an improper dihedral angle
+    - default/generic SMIRKS: `"[*:1]~[*:2](~[*:3])~[*:4]"` 
+
+Here is an example for initiating a chemical environment:
+```python
+from openforcefield.typing.chemistry import environment as env
+smirks = "[#6X3,#7;+0:1]~;@[#8;r:2]~;@[#6X3,#7;+0:3]"
+angle = env.AngleChemicalEnvironment(smirks = smirks)
+```
+
+### ANDtypes and ORtypes
+
+ChemicalEnvironments store information about the SMIRKS string in the same way a 
+chemist might think about a fragment, that is there are atoms connected by bonds. 
+The descriptors for both atoms and bonds are broken down into ORtypes, decorators OR'd together with a ','
+and ANDtypes, decorators AND'd to the end of an atom or bond with a ';'. 
+ORtypes are further broken down into bases and decorators. A base is typically an atomic number or symbol.  
+
+As an example, consider atom 1 above ('[#6X3,#7X2,#7X3;+0:1]'):
+* ORtypes (base, [list of decorators] )
+    - `('#6', ['X3'])`
+    - `('#7', [])`
+* ANDtypes
+    - `['+0']`
+
+You can set and get ORtypes and ANDtypes for atoms and bonds. You can also addORtypes and addANDtypes for atoms and bonds. 
+For example:
+```python
+new_ORtypes = [ ('#6', ['X3']), ('#7', ['X2']) ]
+atom1.setORtypes(new_ORtypes)
+print("New Atom 1: %s " % atom1.asSMIRKS())
+# New Atom 1: [#6X3,#7X2;+0:1]
+
+# Change atom2's AND and OR types with the add*type methods
+atom2 = angle.selectAtom(2)
+atom2.addANDtype('R1')
+atom2.addORtype('#7', ['X3', '+0'])
+print("New Atom 2: %s" % atom2.asSMIRKS())
+# New Atom 2: [#8,#7+0X3;R1:2]
+
+print("\nNew SMIRKS: %s" % angle.asSMIRKS())
+# New SMIRKS: [#6X3,#7X2:1]~;@[#8,#7+0X3;R1:2]~;@[#6X3,#7:3]
+```
+
+### Adding and removing atoms
+
+You can add new atoms to an environment by specifying an atom to connect
+to and the OR and AND types for the new bond and new atom. 
+
+You can also remove existing atoms, this move returns True if the atom was removed and False if it wasn't. 
+Removing an atom removes its associated bonds. Calling removeAtom on an indexed atom or an atom connecting two 
+other atoms will always return False. 
```

Comment:
It returns False and does not remove the atom, there is no exception raised. I'll edit the wording here so that is more clear.